client/blocks/login/social.jsx

```diff
@@ -108,12 +108,20 @@ class SocialLoginForm extends Component {
 	};
 
 	handleAppleResponse = response => {
-		const { onSuccess, redirectTo } = this.props;
+		const { onSuccess, socialService } = this.props;
+		let redirectTo = this.props.redirectTo;
 
 		if ( ! response.id_token ) {
 			return;
 		}
 
+		// load persisted redirect_to url from session storage, needed for redirect_to to work with apple redirect flow
+		if ( socialService === 'apple' && ! redirectTo ) {
+			redirectTo = window.sessionStorage.getItem( 'login_redirect_to' );
+		}
+
+		window.sessionStorage.removeItem( 'login_redirect_to' );
```

Comment:
Can `handleAppleResponse` ever happen on the server? I'm assuming it can't, but otherwise this line (and the similar one above) would need to be guarded.

Reply:
Not sure what you mean by guarded here.
But yeah in theory it would be better to handle the redirect on the server, we actually already do so for the popup flow but it's limited to a single redirect uri while we need to handle many flows and redirects on calypso.

Reply:
I meant doing something along the lines of `typeof window !== 'undefined' && window.sessionStorage...`. If this won't run on the server, though, that's not needed.

Reply:
Oh right 👍 It shouldn't run on the server but I'll double check!